src/components/workoutsRoadmap/services/workoutsRoadmap.service.js

```diff
@@ -3,48 +3,63 @@
 
     angular.module('znk.infra-web-app.workoutsRoadmap').provider('WorkoutsRoadmapSrv', [
         function () {
-            var _newWorkoutGeneratorGetter;
-            this.setNewWorkoutGeneratorGetter = function(newWorkoutGeneratorGetter){
+            var _newSubjectToIgnoreGetter, _newWorkoutGeneratorGetter;
+
+            this.setSubjectToIgnoreGetter = function (newWorkoutGeneratorGetter) {
+                _newSubjectToIgnoreGetter = newWorkoutGeneratorGetter;
+            };
+            this.setNewWorkoutGeneratorGetter = function (newWorkoutGeneratorGetter) {
                 _newWorkoutGeneratorGetter = newWorkoutGeneratorGetter;
             };
 
 
             var _workoutAvailTimesGetter;
-            this.setWorkoutAvailTimes = function(workoutAvailTimesGetter){
+            this.setWorkoutAvailTimes = function (workoutAvailTimesGetter) {
                 _workoutAvailTimesGetter = workoutAvailTimesGetter;
             };
 
-            this.$get = function($injector, $log, $q){
+            this.$get = function ($injector, $log, $q, PersonalizationSrv) {
                 'ngInject';
 
                 var WorkoutsRoadmapSrv = {};
 
-                WorkoutsRoadmapSrv.generateNewExercise = function(subjectToIgnoreForNextDaily, workoutOrder, clickedOnChangeSubjectBtn){
-                    if(!_newWorkoutGeneratorGetter){
-                        var errMsg = 'WorkoutsRoadmapSrv: newWorkoutGeneratorGetter wsa not defined !!!!';
+                WorkoutsRoadmapSrv.generateNewExercise = function (subjectToIgnoreForNextDaily, workoutOrder, clickedOnChangeSubjectBtn) {
+                    var getter, invokedFunc;
+
+                    if (!angular.isFunction(_newWorkoutGeneratorGetter) && !angular.isFunction(_newSubjectToIgnoreGetter)) {
+                        var errMsg = 'WorkoutsRoadmapSrv: getter function was not defined!';
                         $log.error(errMsg);
                         return $q.reject(errMsg);
                     }
-
-                    if(!angular.isArray(subjectToIgnoreForNextDaily)){
+                    if (!angular.isArray(subjectToIgnoreForNextDaily)) {
                         subjectToIgnoreForNextDaily = subjectToIgnoreForNextDaily ? [subjectToIgnoreForNextDaily] : [];
                     }
+                    getter = _newSubjectToIgnoreGetter || _newWorkoutGeneratorGetter;
+                    invokedFunc = $injector.invoke(getter);
 
-                    var newExerciseGenerator = $injector.invoke(_newWorkoutGeneratorGetter);
-                    return $q.when(newExerciseGenerator(subjectToIgnoreForNextDaily,workoutOrder,clickedOnChangeSubjectBtn));
+                    if (_newSubjectToIgnoreGetter) {
```

Comment:
Add a comment - explain that if _newSubjectToIgnoreGetter is defined then we use the new personalziation from  infra, else - support legacy personalization from the web-app.